system_zero/src/alu/subtraction.rs

```diff
@@ -27,5 +57,25 @@ pub(crate) fn eval_subtraction_recursively<F: RichField + Extendable<D>, const D
     yield_constr: &mut RecursiveConstraintConsumer<F, D>,
 ) {
     let is_sub = local_values[IS_SUB];
-    // TODO
+    let in_1 = local_values[COL_SUB_INPUT_0];
+    let in_2 = local_values[COL_SUB_INPUT_1];
+    let out_1 = local_values[COL_SUB_OUTPUT_0];
+    let out_2 = local_values[COL_SUB_OUTPUT_1];
+    let out_br = local_values[COL_SUB_OUTPUT_BORROW];
+
+    let base = builder.constant_extension(F::Extension::from_canonical_u64(1 << 16));
+    let base_sqr = builder.constant_extension(F::Extension::from_canonical_u64(1 << 32));
+
+    // rhs = out_1 + base * out_2
+    let rhs = builder.mul_add_extension(out_2, base, out_1);
```

Comment:
We generally try to make the two eval functions (native & recursive) very similar, so it's easy to audit one and then check that the other matches. A couple nits to increase similarity:
- Let's move `rhs` below `lhs`
- Maybe inline `out_diff` in the native version to match this version